main.js

```diff
@@ -165,27 +193,149 @@ app.use("/images", express.static("public/images"));
 app.use("/html", express.static("public/html"));
 
 app.get('/', function (req, res) {
-	if (req.session.loggedIn) {
-		res.redirect('/profile');
-		console.log("User logged in!");
-		return;
-	}
+	
 	let doc = fs.readFileSync("./html/index.html", "utf8");
-	res.send(doc);
+	let index = new JSDOM(doc);
+
+	// Add the footer
+	index = loadHTMLComponent(index, "footer", "footer", "./html/footer.html");
+
+	res.send(index.serialize());
 });
+app.get('/profile-details', async(req, res) => {
+	const filterQuery = {emailAddress: req.session.email};
+
+	const userResults = await db.collection('BBY-6_users').find(filterQuery).toArray();
+	if(userResults.length === 0) {
+		// Could not find user
+		res.status(500).send("userNotFound");
+		return;
+	} else {
+		res.setHeader("Content-Type", "application/json");
+		res.status(200).send({
+			name: req.session.name,
+			email: req.session.email
+		})
+	}
 
-app.get('/profile', function (req, res) {
+})
+app.get('/profile', async (req, res) => {
 	if (!req.session.loggedIn) {
 		res.redirect("/login");
 		return;
 	}
-	let doc = fs.readFileSync("./html/profile.html", "utf8");
-	res.send(doc);
+	let doc = fs.readFileSync("./html/user-profile.html", "utf8");
+	const baseDOM = new JSDOM(doc);
+	let profile = loadHeaderFooter(baseDOM);
+
+	profile = loadHTMLComponent(profile, "#Base-Container", "div", "./html/profile-component.html");
+    profile.window.document.getElementById("FullName").defaultValue = req.session.name;
+	profile.window.document.getElementById("Email").defaultValue = req.session.email;
+	res.send(profile.serialize());
 });
 
+app.patch('/profile', async(req, res) => {
+	
+
+	// if the request is not coming from a logged in user, reject.
+	if (!req.session || !req.session.loggedIn) {
+		res.status(401).send("invalidSession");
+		return;
+	} 
+	
+	// get user from db
+	const filterQuery = {emailAddress: req.session.email};
+	const userResults = await db.collection('BBY-6_users').find(filterQuery).toArray();
```

Comment:
Wonder if it's better to skip this, and just do a single request to the database, catch this error case on the update failing. Reasoning being that two requests are slower than one, and that there's actually a weird race condition here, if someone deletes a user you want to update, just after you run the "user exists" query.

Reply:
Tracked in #37, resolving this for now.